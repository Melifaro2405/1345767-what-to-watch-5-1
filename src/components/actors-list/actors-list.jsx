import React from "react";
import PropTypes from "prop-types";

export const ActorsList = ({actors}) => {
  return (
    <ul style={{listStyleType: `none`, padding: 0, margin: 0}}>
      {actors.map((actor, i) =>
        <li key={`${i}-${actor}`}>
          {actor},
        </li>)}
    </ul>
  );
};

ActorsList.propTypes = {
  actors: PropTypes.arrayOf(PropTypes.string).isRequired
};
