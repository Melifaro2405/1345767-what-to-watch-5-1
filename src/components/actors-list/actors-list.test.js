import React from 'react';
import renderer from 'react-test-renderer';
import {ActorsList} from './actors-list';

it(`Should ActorsList render correctly`, () => {
  const tree = renderer
    .create(<ActorsList
      actors={[`test1, test2, test3`]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
