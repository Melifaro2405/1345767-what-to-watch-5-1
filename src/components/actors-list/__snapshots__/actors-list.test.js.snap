// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ActorsList render correctly 1`] = `
<ul
  style={
    Object {
      "listStyleType": "none",
      "margin": 0,
      "padding": 0,
    }
  }
>
  <li>
    test1, test2, test3
    ,
  </li>
</ul>
`;
